src/rust/catcollar/mod.rs

```diff
@@ -385,46 +352,35 @@ impl CatcollarLibOS {
     /// Closes a socket.
     pub fn close(&mut self, qd: QDesc) -> Result<(), Fail> {
         trace!("close() qd={:?}", qd);
-        let mut qtable: RefMut<IoQueueTable<CatcollarQueue>> = self.qtable.borrow_mut();
-        match qtable.get(&qd) {
-            Some(queue) => match queue.get_fd() {
-                Some(fd) => match unsafe { libc::close(fd) } {
-                    stats if stats == 0 => (),
-                    _ => {
-                        let errno: libc::c_int = unsafe { *libc::__errno_location() };
-                        error!("failed to close socket (fd={:?}, errno={:?})", fd, errno);
-                        return Err(Fail::new(errno, "operation failed"));
-                    },
-                },
-                None => unreachable!("CatcollarQueue has invalid underlying file descriptor"),
+        let fd: RawFd = self.get_queue_fd(&qd)?;
+        match unsafe { libc::close(fd) } {
+            stats if stats == 0 => {
+                self.qtable
+                    .borrow_mut()
+                    .free::<CatcollarQueue>(&qd)
+                    .expect("queue should exist");
```

Comment:
To follow out guidelines, could you add a comment here to explain why it is safe to call `expect()` here?